src/config.ts

```diff
@@ -19,56 +19,250 @@ export const DEFAULT_CONFIG = `
+    private static instance?: MirrordConfigManager = undefined;
+
+    /**
+     * Active config. User can set this for the whole workspace.
+     */
+    private active?: vscode.Uri;
+    private fileListener: vscode.Disposable;
+
+    private constructor() {
+        this.fileListener = vscode.workspace.onDidDeleteFiles(event => {
+            event.files.find(file => {
+                if (file.path === this.active?.path) {
+                    vscode.window.showWarningMessage("removed active mirrord configuration");
+                    this.active = undefined;
+                }
+            });
+        });
+    }
+
+    public dispose() {
+        this.fileListener.dispose();
+    }
+
+    /**
+     * @returns a global instance of this manager
+     */
+    public static getInstance(): MirrordConfigManager {
+        if (MirrordConfigManager.instance === undefined) {
+            MirrordConfigManager.instance = new MirrordConfigManager();
+        }
+
+        return MirrordConfigManager.instance;
     }
-    vscode.workspace.openTextDocument(path).then(doc => {
+
+    /**
+     * Handles `mirrord.selectActiveConfig` command.
+     * Allows the user to set an active mirrord config from quick pick.
+     * Any path across the workspace is available, as long as its name ends with `mirrord.{json,toml,yml,yaml}`.
+     */
+    public async selectActiveConfig() {
+        const options: Map<string, vscode.Uri> = new Map();
+        const files = await vscode.workspace.findFiles("**/*mirrord.{json,toml,yml,yaml}");
+        files.forEach(f => options.set(`use ${vscode.workspace.asRelativePath(f)}`, f));
+
+        const displayed = ["<none>", ...options.keys()];
+        const placeHolder = this.active
+            ? `Select active mirrord config from the workspace (currently ${vscode.workspace.asRelativePath(this.active)})`
+            : "Select active mirrord config from the workspace";
+        const selected = await vscode.window.showQuickPick(displayed, {placeHolder});
+        if (selected === "<none>") {
+            this.active = undefined;
+        } else if (selected) {
+            let path = options.get(selected)!!;
+            this.active = path;
+        }
+    }
+
+    /**
+     * Searches the given workspace folder for a default config.
+     * Default configs are located in the `.mirrord` directory and their names end with `mirrord.{toml,json,yml,yaml}`.
+     * If there are multiple candidates, they are sorted according alphabetically and the first one is returned.
+     * @param folder searched workspace folder
+     * @returns path to the found config
+     */
+    private static async getDefaultConfig(folder: vscode.WorkspaceFolder): Promise<vscode.Uri | undefined> {
+        let pattern = new vscode.RelativePattern(folder, ".mirrord/*mirrord.{toml,json,yml,yaml}");
+        let files = await vscode.workspace.findFiles(pattern);
+        files.sort((f1, f2) => {
+            if (f1.path > f1.path) {
```

Comment:
```suggestion
            if (f1.path > f2.path) {
```

Reply:
Also, I think you can replace this whole block with `f1.path.localeCompare(f2.path)`